source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -79,27 +81,58 @@ MStatus bindFileTexture(MHWRender::MShaderInstance& shader,
     MHWRender::MTextureManager* textureManager = renderer ? textureManager = renderer->getTextureManager() : nullptr;
     if (textureManager)
     {
-        MaterialX::FilePath imagePath = MaterialXMaya::findInSubdirectories(searchPath, fileName);
-        if (!imagePath.isEmpty())
+        MTexture *texture = nullptr;
+        if (udimSet && !udimSet->empty())
         {
+            std::vector<int> udimTiles;
+            mx::FilePathVec udimPaths = mx::ImageHandler::getResolveUDIMInformation(fileName, *udimSet, udimTiles);
+            if (udimTiles.size() != udimPaths.size() * 2)
```

Comment:
This check looks mysterious with the current API, so I'm really proposing returning a vector of custom structs instead - see comments around `getResolveUDIMInformation`.